catalog/transformer/catalog-transformer-service-atom/src/main/java/ddf/catalog/transformer/response/query/atom/AtomTransformer.java

```diff
@@ -473,4 +478,24 @@ private Link addLinkHelper(Action action, Entry entry, String linkType, String m
     }
     return georssPositions;
   }
+
+  private int getCount(SourceResponse queryResponse) {
+    if (queryResponse == null) {
+      return -1;
+    }
+
+    QueryRequest queryRequest = queryResponse.getRequest();
+
+    if (queryRequest == null) {
+      return -1;
+    }
+
+    Serializable countSerializable = queryRequest.getProperties().get("count");
+
+    if (countSerializable == null) {
+      return -1;
+    }
+
+    return Integer.valueOf((String) countSerializable);
```

Comment:
This can throw a `NumberFormatException` if `countSerializable` does not conform to a proper decimal integer format. This should be surrounded by a `try`-`catch` block and that case should be explicitly handled.

Reply:
Good catch.  I've wrapped this in a try/catch.